compiler/zrc/src/compile.rs

```diff
@@ -18,8 +24,11 @@ pub fn compile(
     triple: &zrc_codegen::TargetTriple,
     cpu: &str,
 ) -> Result<Box<[u8]>, zrc_diagnostics::Diagnostic> {
+    // === PREPROCESSOR ===
+    let chunks = zrc_preprocessor::preprocess(Path::new(parent_directory), file_name, content)?;
+
     // === PARSER ===
-    let ast = zrc_parser::parser::parse_program(content)?;
+    let ast = zrc_parser::parser::parse_source_chunks(&chunks)?;
```

Comment:
Make the parser handle only parsing *one* atomic source chunk. Then, handle all the mutability in the driver.